### PR TITLE
Release 0.31

### PR DESCRIPTION
**Pull request by akaszynski**
### Release 0.31

It's only three weeks since our last release, but we've added several features and bug fixes that would be great to get out as a release, including:

- Themes
- Patching scalar bar removal
- Improved streamline examples
- Patched screenshot generation
- Option to use cell data within glyph filter
- Perlin noise examples
- Improved testing
- Faster ray tracing
- Ensure modifying arrays updates the render pipeline
- Improve build notes

Additionally, with this release we're also going to add our own wheels in the release notes, including aarch64 build for with and without EGL, EGL wheels, and python 3.9 (all built correctly and around 100 MB).

#### Misc notes
As always, this release branch (when merged to master) will bump the version to the next minor version with the `dev0` tag.  We'll keep this release branch and bump to `0.31.0`.

This PR also adds a few badges to our [README.rst](https://github.com/pyvista/pyvista/blob/release/0.31/README.rst).

I'll be clearing out many of the older branches (release and otherwise) since there's no reason to maintain minor versions over 6 months old.  We've yet to backport a patch to an older release, but we'll still keep the past two or three around just in case.

### CODE REVIEW

**Comment by adeak** [comment]
If I understand correctly, the VTK wheels will be available as artifacts on the release page on github (not sure I got the terminology right). I assume these don't get built automatically, so perhaps their release should be added to [the minor release documentation](https://github.com/pyvista/pyvista/blob/master/CONTRIBUTING.md#minor-release-steps). Or whatever version level we want to publish VTK wheels with.

**Comment by akaszynski** [comment]
Since we're still providing unofficial wheels, I don't think formalizing it in the docs is necessary, nor will it be built for every release.  The point of adding wheels to the release notes is to have somewhere where we can post binaries for free.  I could make a pyvista PyPi server for fun, but it seems easier just to put them in the release notes for those who just want wheels.

Suggestions?

**Comment by adeak** [comment]
I see. So the new "Building VTK extras" page that says
> Should you need a prebuilt wheel, a variety of prebuilt wheels (including ones with EGL) can be found at pyvista Release 0.31.0. This includes wheels for Raspberry Pi (64-bit OS), Python 3.9 Linux, and EGL support (arch64 and x86_64).
is meant to be a fixed link rather than something that gets bumped as new wheels get built for future VTK versions.

I think that's fine too, I just misunderstood the intent. And a private PyPI server might be overkill.

**Comment by akaszynski** [comment]
> is meant to be a fixed link rather than something that gets bumped as new wheels get built for future VTK versions.

Correct.  Would be great if VTK can start building wheels, especially as not having VTK on Python 3.9 is a deal breaker in many cases where users just download the latest Python.

**Comment by akaszynski** [comment]
Would like to get approval for this.  Please let me know if there's any issues with this release.

File: README.rst
Comment (adeak) [inline README.rst:47]: Are the PyPI and conda-forge links unused? Were they meant to be `:target:`s instead?

File: docs/index.rst
Comment (adeak) [inline docs/index.rst:235]: Same as above.

**Review by adeak — COMMENTED** [review]
I remembered the detailed release notes from https://github.com/pyvista/pyvista/pull/1224 and I assumed you weren't done here.

One question in 4 places.

**Comment by adeak** [comment]
Not an issue _per se_, but I started a little work on weeding out remaining user-facing assertions. It's not a huge deal, and it shouldn't change behaviour for the worse, so it doesn't have to go in this release. I can only finish that tonight the earliest, and I don't want to hold up the release.

File: docs/index.rst
Comment (akaszynski) [reply docs/index.rst]: Fixed.  Thanks for finding that.

File: README.rst
Comment (akaszynski) [reply README.rst]: Fixed

**Review by adeak — APPROVED** [review]

**Comment by akaszynski** [comment]
Thanks @adeak

**Pull request merged**
